File: enum/2975移除栅栏得到的正方形田地的最大面积.cpp
#include <unordered_set>
#include <vector>
#include <algorithm>
using namespace std;
class Solution {
private:
    unordered_set<int>f(vector<int>&a,int mx){
        a.push_back(1);
        a.push_back(mx);
        sort(a.begin(),a.end());
        unordered_set<int>res;
        for(int i=0;i<a.size();i++){
            for(int j=i+1;j<a.size();j++){
                //cout<<a[j]-a[i]<<endl;
                res.insert(a[j]-a[i]);
            }
        }
        return res;
    }
    const int MOD = 1000000007;
public:
    int maximizeSquareArea(int m, int n, vector<int>& hFences, vector<int>& vFences) {
        unordered_set<int>h=f(hFences,m);
        unordered_set<int>v=f(vFences,n);
        int ans=0;
        for(int x:h){
            if(v.count(x)){
                ans = max(x,ans);
            }
        }
        if(ans==0)return -1;
        return 1LL*ans*ans % MOD;
    }
};

File: enum/2975移除栅栏得到的正方形田地的最大面积.py
from itertools import combinations
from typing import List
class Solution:
    def f(self,a: List[int],mx: int) -> int:
        a +=[1, mx]
        a.sort()
        return set(y - x for x, y in combinations(a, 2))
    def maximizeSquareArea(self, m: int, n: int, hFences: List[int], vFences: List[int]) -> int:
        MOD = 1000000007
        h_set = self.f(hFences, m)
        v_set = self.f(vFences, n)
        ans = max(h_set & v_set,default = 0)
        return ans*ans % MOD if ans else -1